components/scream/src/physics/shoc/shoc_pblintd_height_impl.hpp

```diff
@@ -52,7 +52,7 @@ void Functions<S,D>::pblintd_height(
   const Int upper = nlev-1;
   const Int length = upper-lower;
   const Int lower_pack_indx = lower/Spack::n;
-  const Int upper_pack_indx = lower_pack_indx + (length+Spack::n-1)/Spack::n;
+  const Int upper_pack_indx = lower_pack_indx + (length+Spack::n)/Spack::n;
```

Comment:
Doesn't this make it so that there's always at least one iteration? The interval [L,U) , with U>=L+1, always runs at least once...

Reply:
But maybe that's the point? Is upper supposed to be "the last" level to be run on?

Reply:
Ahh yeah it does. Ok, so there has to be a way to avoid that. But the previous index was one short (I think always). I mean, we could add a check if `upper <= lower`.

Reply:
@bartgol So I think the original issue was not that we were executing the `parallel_reduce()` when we shouldn't, but that when we did execute and never found an `if (inrange.any())`, then `max_indx = Kokkos::recuct_id<Int>::max()` was returned instead of the `-1` we were comparing against. I think we could run the `parallel_reduce` over all levels and still get the correct answer since we have the `Mask` checking for `in_range.any()`. 
As long as we use ` Kokkos::recuct_id<Int>::max()`

I think the other issue is unrelated, so you could test this out in your restart branch.

Reply:
So it turns out the new indexing was still wrong in cases with 16 pack size. I reverted back to the old way, but added an `if (u==l) return` so that we don't reduce when there is nothing to reduce. I do still believe that is ok to do, but might as well skip the work.

Reply:
So, you want to run the reduction on all levels, regardless? That might be ok on GPU, but it might be suboptimal on CPU. That said, correctness is the first concern, so whatever gets the correct results, can be merged, and performance be tackled with a follow up PR.

Reply:
nvm, I looked at the code,, and it seems we do the ||reduce only on the packs we need, so it's fine. Thanks!